docs/Custom_Device_Tips_and_Tricks.md

```diff
@@ -0,0 +1,271 @@
+
+Consider the following example line from the *Custom Device.xsd* file:
+
+<xs:element minOccurs="0" name="ActionVIOnDelete" type="Path" />
+
+**A Line from the Custom Device XML Schema File**
+
+The name of this tag is *ActionVIOnDelete*. Adding this tag to the custom device XML runs a VI when the operator deletes the item from System Explorer. You may find experimenting with the custom device XML easier in an empty custom device.
+
+It may be helpful to explore VeriStand’s built-in components for examples on implementing XML features. The built-in components are found in &lt;Application Data&gt;\System Explorer\System Explorer Definition Files.
+
+If a tag is opened, use the format *</tag_name>* to close the tag. If a tag must be specified but has no value, you may use the format *<tag_name />* to open and close the tag at the same time. This format has the same effect as *<tag_name>tag value</tag_name>*.
+
+### Delete Protection
+
+You can add &lt;DeleteProtection&gt;true&lt;/DeleteProtection&gt; to any section in the [custom device XML](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-xml-tags/) to disallow deleting the item from the configuration tree in System Explorer.
+
+### Limiting Occurrences of the Custom Device
+
+If it doesn’t make sense to have more than **N** instances of the custom device in a single System Definition, you can limit the number of instances by adding **&lt;MaxOccurrence&gt;N&lt;/MaxOccurrence&gt;** to the [custom device XML](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-xml-tags/) underneath the device type.
+
+### Rename Protection
+
+There may be cases when you depend on a custom device item to have a certain name, and you’d like to prevent the operator from renaming the item. Add **&lt;DisallowRenaming*gt;true&lt;/DisallowRenaming&gt;** below the &lt;/Name&gt; tag for any page to prevent the operator from renaming the item.
+
+### Action VIs
+
+VeriStand contains eight action VI templates that are triggered by different actions.
+
+The following list displays the action VI templates provided by VeriStand in the **[Custom Device API library](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-api-library/)**.
+
+**ActionVIOnLoad**
+Executes when VeriStand loads a custom device item into memory. This template helps create action VIs that launch background processes. For example, if your custom device requires large amounts of data, you can customize this template to start a daemon that runs processes or gathers data in the background.
+
+**ActionVIOnDeleteRequest**
+Executes when a user tries to delete an item from the custom device. This template helps create action VIs that prevent a user from deleting a custom device item or warn a user of the implications of deleting a custom device item.
+
+The template has the following unique parameters.<br />
```

Comment:
All the template details below should be indented, for all the template VIs that have lists of this type.